### DISCUSSION TITLE
export/pack nicgui page into vue component

### DISCUSSION THREAD

**Discussion by ultranity** [zauberzeug/nicegui#1697] [Ideas / Feature Requests]
Nice Project!

I want to implement some new function with nicegui into our current Vue project, so I am looking for some way to make it competible with current frontend pages (better if we can use it in vue-router).

Therefore, I am wondering if we can export or pack nicgui page into vue component, so that it could be integrate into other frontend projects? this could also help to debug frondend issues.

Or if there is any alternative suggestion/solution to mount nicgui page to other vue projects?

**Comment by falkoschindler** [comment]
NiceGUI follows a backend-first philosophy: The app is basically a FastAPI backend serving HTML pages that are updated via WebSockets. Therefore it's pretty easy to [integrate NiceGUI into an existing FastAPI app](https://github.com/zauberzeug/nicegui/tree/main/examples/fastapi). But I don't see how to use it with an existing Vue frontend. The only way I can imagine would be to load a more or less independent NiceGUI app in an iframe. But maybe someone else has an idea?
